WordPress/src/main/java/org/wordpress/android/ui/posts/GutenbergWarningFragmentDialog.kt

```diff
@@ -0,0 +1,135 @@
+package org.wordpress.android.ui.posts
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.support.v7.app.AppCompatDialogFragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.CheckBox
+import org.wordpress.android.R
+import org.wordpress.android.widgets.WPTextView
+
+/**
+ * Basic dialog fragment with support for 1,2 or 3 buttons.
+ */
+class GutenbergWarningFragmentDialog : AppCompatDialogFragment() {
+    private lateinit var mTag: String
+    private var mGutenbergRemotePostId: Long = 0
+    private var mIsPage: Boolean = false
+
+    interface GutenbergWarningDialogClickInterface {
+        fun onGutenbergWarningDialogEditPostClicked(instanceTag: String, gutenbergRemotePostId: Long)
+        fun onGutenbergWarningDialogCancelClicked(instanceTag: String, gutenbergRemotePostId: Long)
+        fun onGutenbergWarningDialogLearnMoreLinkClicked(instanceTag: String, gutenbergRemotePostId: Long)
+        fun onGutenbergWarningDialogDontShowAgainClicked(
+            instanceTag: String,
+            gutenbergRemotePostId: Long,
+            checked: Boolean
+        )
+    }
+
+    fun initialize(
+        tag: String,
+        isPage: Boolean,
+        gutenbergRemotePostId: Long
+    ) {
+        mTag = tag
+        mIsPage = isPage
+        mGutenbergRemotePostId = gutenbergRemotePostId
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        this.isCancelable = false
+        val theme = 0
+        setStyle(AppCompatDialogFragment.STYLE_NORMAL, theme)
+
+        if (savedInstanceState != null) {
+            mTag = savedInstanceState.getString(STATE_KEY_TAG)
+            mIsPage = savedInstanceState.getBoolean(STATE_KEY_IS_PAGE)
+            mGutenbergRemotePostId = savedInstanceState.getLong(STATE_KEY_GUTENBERG_REMOTE_POST_ID)
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putString(STATE_KEY_TAG, mTag)
+        outState.putBoolean(STATE_KEY_IS_PAGE, mIsPage)
+        outState.putLong(STATE_KEY_GUTENBERG_REMOTE_POST_ID, mGutenbergRemotePostId)
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.gutenberg_warning_dialog, container, false)
+        initializeView(view)
+        return view
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState)
+        dialog.setCanceledOnTouchOutside(false)
+        dialog.setCancelable(false)
```

Comment:
Since canceling the dialog doesn't seem to have side effects to the flow, do we need to be so strict about its cancellability (both on tap-outside and via the hardware Back button)?

Reply:
I think it's better to make sure the flows are actively enforced and tracked in this particular case. Also, given they want to open a Post, would canceling (by touching elsewhere) mean "get out of my way, open the Post", or "ok no let's go back"?.
I prefer the unambiguity here.